CMakeLists.txt

```diff
@@ -267,16 +267,23 @@ endif()
 # Make sure the generated paths exist
 execute_process(COMMAND mkdir -p "${CMAKE_BINARY_DIR}/generated")
 
-# We need to link some packages as dynamic/dependent.
-set(CMAKE_FIND_LIBRARY_SUFFIXES .dylib .so)
-find_package(OpenSSL REQUIRED)
-find_package(BZip2 REQUIRED)
-find_package(Dl REQUIRED)
-find_package(Readline REQUIRED)
+# Apple has deprecated OpenSSL, static link with brew bottled libressl
+if(APPLE)
+  set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
```

Comment:
Right! An else that just sets shared! 

```cmake
if(NOT DEFINED ENV{BUILD_LINK_SHARED})
  set(CMAKE_FIND_LIBRARY_SUFFIXES .a .dylib .so)
else()
  set(CMAKE_FIND_LIBRARY_SUFFIXES .dylib .so)
endif()
```

The define is meant for building packages within a package manager. It should affect all dependencies, in that case brew will pull in the RocksDB shared library and link against it. The formula update can wait until the next release for sure.